packages/odyssey-react-mui/src/Select.tsx

```diff
@@ -345,7 +348,7 @@ const Select = <
                     key={item}
                     label={
                       <>
-                        {item}
+                        {normalizedOptionsMap.get(item)?.text}
```

Comment:
I thought about this for a while. Why use a map instead of an object or an array. I think the correct solution is to use `normalizedOptions` instead of `internalSelectedValues`. Then we don't have to do these lookups.

The performance of n^2 could get messed up with 2K+ items, but probably not. Either way, I'm fine with using a different data type, I'm just wondering if it's better overall to rewrite the logic, so we don't need that in the first place.

What about taking this code and doing the heavy lifting there?:
```ts
  useEffect(() => {
    if (controlledStateRef.current === CONTROLLED) {
      setInternalSelectedValues(value);
    }
  }, [value]);
  ```
  
So instead of `setInternalSelectedValues`, it's `setInternalSelectedOptions`, and we do the lookup at that point. The cost of a lookup would be much faster if we only do it for a few values than having to do it each render cycle.

Also, we wouldn't have to maintain another dataset which could be confusing.

Lastly, it might be useful to make the options list a map in the first place. Then we could do `Array.from(map.entries())` anytime we want those values.

We could avoid something like this:
```ts
setInternalSelectedOptions(internalSelectedOptions => (
  internalSelectedOptions
  .filter((internalSelectedOption) => (
    selectedValues.includes(internalSelectedOption.value)
  ))
))
```

But either way works for me. You'd be surprised how fast they execute even on slower devices. By moving the grunt-work logic to the point of selection, you remove a lot of the continual n^2 execution time.

While wordy, my point is, I'd prefer 1 data type, not 2. Pick the one that makes the most sense and write the rest of the code around that.